api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -42,4 +42,13 @@
    * @return this for method chaining
    */
   RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> filesToAdd);
+
+  /**
+   * Add a rewrite that replaces one set of deletes with another that contains the same deleted rows.
+   *
+   * @param deletesToDelete files that will be replaced, cannot be null or empty.
+   * @param deletesToAdd files that will be added, cannot be null or empty.
+   * @return this for method chaining
+   */
+  RewriteFiles rewriteDeletes(Set<DeleteFile> deletesToDelete, Set<DeleteFile> deletesToAdd);
```

Comment:
Before we start the replacing equality deletes with position deletes,  I think we need to refactor the RewriteFiles API   to adjust more cases: 
1.   Rewrite data files and remove all the delete rows.  The files to delete will be a set of data files and a set of delete files, and the files to add will be a set of data files.
2. Replace equality deletes with position deletes,  the files to delete will be a set of equality delete files (we will need to ensure that all delete files are equality delete files ? ) ,  the files to add will be a set of position delete files.
3.  Merging small delete files into a bigger delete files.  The files to delete will be  a set of equality/position delete files,  the files to add will be a set of equality/position delete files.

Reply:
That makes sense to me.  I think we could parallelize the API refactoring and the implementation.